### DISCUSSION TITLE
How do I configure monaco-languageclient to enable Client Capabilities for python LSP?

### DISCUSSION THREAD

**Discussion by Taco7** [TypeFox/monaco-languageclient#463] [Q&A]
I am trying to hookup a monaco editor with a python lsp server. Currently using  [Palantir's LSP](https://github.com/palantir/python-language-server) along with the [sample client ](https://github.com/TypeFox/monaco-languageclient/tree/main/packages/examples/client).

The only feature which seems to be working as of now are the  [code suggestions](https://imgur.com/UjFfB4l) but additional features like hover and jump to definition are not. The client is not able to process the response given by the server. I am attaching a snippet of the network messages, but I am not sure how do I resolve this. [Request-Response](https://imgur.com/TlhFrHc)

**Comment by CGNonofr** [comment]
You probably need your file to exist on the filesystem, and not use "inmemory://" scheme

**Reply by ghost** [reply]
Why is the "inmemory://" scheme not possible

**Reply by CGNonofr** [reply]
It is expected that the language server process has access to the same file system as the client.

For instance, the client sends the textDocument/didSave notification, and the server needs to have access to the filesystem to get the new value of the file
